include/linux/soundwire/sdw.h

```diff
@@ -650,10 +650,7 @@ struct sdw_link_ops;
 struct sdw_master_device {
 	struct device dev;
 	struct sdw_bus *bus;
-	struct sdw_link_ops *link_ops;
-	int link_id;
 	bool pm_runtime_suspended;
-	void *pdata;
```

Comment:
I think it would make reviewing this patch easier if you could split it into several. The difficulty is, that this patch seems to be doing several things: (1) removing several fields out of `struct sdw_master_device` and switching over to using those from `struct sdw_bus`, adding them there if needed, (2) performing various functional changes. Maybe it could be split at least into: (1) move `.link_ops`, (2) move `.link_id`, (3) move `pdata`, (4) do the rest?

Reply:
> I think it would make reviewing this patch easier if you could split it into several. The difficulty is, that this patch seems to be doing several things: (1) removing several fields out of `struct sdw_master_device` and switching over to using those from `struct sdw_bus`, adding them there if needed, (2) performing various functional changes. Maybe it could be split at least into: (1) move `.link_ops`, (2) move `.link_id`, (3) move `pdata`, (4) do the rest?

@lyakh #2051 Should make it easier :)